quinn-proto/src/assembler.rs

```diff
@@ -86,12 +86,17 @@ impl Assembler {
         let new = BinaryHeap::with_capacity(self.data.len());
         let old = mem::replace(&mut self.data, new);
         for chunk in old.into_sorted_vec().into_iter().rev() {
-            if offset + (buffer.len() as u64) < chunk.offset {
+            let end = offset + (buffer.len() as u64);
+            let overlap = end.saturating_sub(chunk.offset);
+            if end - overlap != chunk.offset {
```

Comment:
I find this difficult to follow. Could it be written without comparing values that were derived from eachother? Perhaps `if end < chunk.offset`?